src/Components/test/E2ETestMigration/Tests/BindTest.cs

```diff
@@ -0,0 +1,1367 @@
+using PlaywrightSharp;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Tests
+{
+    public class BindTest : ServerTestBase<ToggleExecutionModeServerFixture<Program>>
+    {
+        public BindTest(
+            ToggleExecutionModeServerFixture<Program> serverFixture,
+            ITestOutputHelper output)
+            : base(serverFixture, output)
+        {
+        }
+
+        protected override Type TestComponent { get; } = typeof(BindCasesComponent);
+
+        protected override async Task InitializeCoreAsync(TestContext context)
+        {
+            await base.InitializeCoreAsync(context);
+
+            // On WebAssembly, page reloads are expensive so skip if possible
+            //Navigate(ServerPathBase, noReload: _serverFixture.ExecutionMode == ExecutionMode.Client);
+        }
+
+        protected async Task<string> GetInputValue(string selector)
+            => await TestPage.EvalOnSelectorAsync<string>(selector, "e => e.value");
+
+        [Theory]
+        [InlineData(BrowserKind.Chromium)]
+        [InlineData(BrowserKind.Firefox)]
+        [InlineData(BrowserKind.Webkit)]
+        // NOTE: BrowserKind argument must be first
+        public async Task CanBindTextbox_InitiallyBlank(BrowserKind browserKind)
+        {
+            if (ShouldSkip(browserKind)) 
+            {
+                return;
+            }
+
+            var target = await TestPage.QuerySelectorAsync("#textbox-initially-blank");
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank"));
+            Assert.Equal(string.Empty, await TestPage.GetInnerTextAsync("#textbox-initially-blank-value"));
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank-mirror"));
+
+            // Modify target; verify value is updated and that textboxes linked to the same data are updated
+            await target.TypeAsync("Changed value");
+            // Doesn't update until change event
+            Assert.Equal(string.Empty, await TestPage.GetInnerTextAsync("#textbox-initially-blank-value"));
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank-mirror"));
+            await target.PressAsync("Tab");
+            Assert.Equal("Changed value", await TestPage.GetInnerTextAsync("#textbox-initially-blank-value"));
+            Assert.Equal("Changed value", await GetInputValue("#textbox-initially-blank-mirror"));
+
+            // Remove the value altogether
+            await TestPage.ClickAsync("#textbox-initially-blank-setnull");
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank"));
+            Assert.Equal(string.Empty, await TestPage.GetInnerTextAsync("#textbox-initially-blank-value"));
+            Assert.Equal(string.Empty, await GetInputValue("#textbox-initially-blank-mirror"));
+        }
+
+        //[Fact]
+        //public void CanBindTextbox_InitiallyPopulated()
```

Comment:
Are you wanting to convert these tests before merging the PR or?

Reply:
No, I just want a small set of tests to start with, there's 12 right now.  I don't want to onboard too many tests before we are happy and settle on the overall pattern.  Its already a bit of a hassle to have to update so many tests when changing some of test patterns.

I had some open questions still about whether it was safe or a good idea to reuse the browser navigation page (TestPage property), I had to do some slightly sketchy stuff to make this work with the BrowserKind attribute.

https://github.com/dotnet/aspnetcore/blob/328910d83b56127165e94cc6e3db67ea799d154c/src/Components/test/E2ETestMigration/Infrastructure/ComponentTestBase.cs#L50

I also wanted to make sure we were happy with the existing MountTestComponent pattern before going too much farther.

https://github.com/dotnet/aspnetcore/blob/328910d83b56127165e94cc6e3db67ea799d154c/src/Components/test/E2ETestMigration/Infrastructure/ComponentTestBase.cs#L25

This is the first time I've seen how these test were setup, but now is a good time to consider if there's any improvements to make here.  I don't know have anything in mind in terms of changes or improvements, but I wanted to double check we are happy with this method of test selection

Reply:
In terms of next steps, I think once we are mostly happy with the shape of  things, I'll probably try turning on a single test in every file to play minesweeper and see what explodes and find where the gaps still are (in a future PR)